### ISSUE TITLE
Can't commit (with certain config)

### ISSUE THREAD

**Issue by wrightjjw** [NeogitOrg/neogit#563]
### Description

On Windows and Fedora, I can open the commit dialog and put in a commit, but there is no syntax highlighting, and writing the commit dialog does not create a commit.

On WSL the commit dialog doesn't even open...

**Important:** I've narrowed down this problem to probably being related to some plugin I have installed, because running with the minimal config works fine. Thought I'd still open this anyway to see if I can track down a potential fix. [This](https://github.com/wrightjjw/neovim-config/blob/main/lua/plugins.lua) is my plugin config if anyone is crazy enough to look into it themselves.

### Neovim version

NVIM v0.9.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=auto -fstack-protector-strong -DUNIT_TESTING -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/usr/include/luajit-2.1 -I/usr/include -I/usr/include/luv -I/builddir/build/BUILD/neovim-0.9.1/redhat-linux-build/src/nvim/auto -I/builddir/build/BUILD/neovim-0.9.1/redhat-linux-build/include -I/builddir/build/BUILD/neovim-0.9.1/redhat-linux-build/cmake.config -I/builddir/build/BUILD/neovim-0.9.1/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include

### Operating system and version

Fedora 38

### Steps to reproduce

Stage commits, then `cc`

### Expected behavior

commit works

### Actual behavior

it doesn't

### Minimal config

```Lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])
local package_root = "/tmp/nvim/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup({
    {
      "wbthomason/packer.nvim",
      {
        "TimUntersberger/neogit",
        requires = {
          { "nvim-lua/plenary.nvim" },
          { "sindrets/diffview.nvim" },
        },
        config = function()
          print("loaded neogit")
          require("neogit").setup()
        end,
      },
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  })
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing neogit and dependencies.")
  vim.fn.system({ "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path })
end
load_plugins()
require("packer").sync()
```

**Comment by ten3roberts** [comment]
Thanks for your issue.

If I may ask, is this before or after @CKolkey's merge?

**Comment by wrightjjw** [comment]
Only after!

**Comment by cuducos** [comment]
Same here, no commit, no error after the big merge (BTW congrats for tons of improvement, folks!).

I may try to find what plugins @wrightjjw and I have in common ([mine](https://github.com/cuducos/dotfiles)).

**Comment by CKolkey** [comment]
I wonder - do you have the `gitcommit` language installed with tree-sitter?

**Comment by wrightjjw** [comment]
> I wonder - do you have the `gitcommit` language installed with tree-sitter?

I did not... Installing it fixed it in Fedora but not in Windows. Still no highlighting in Windows either...

**Comment by skoch13** [comment]
Installing `gitcommit` helps. Thank you @CKolkey

**Comment by CKolkey** [comment]
That _should_ be an optional upgrade, not a requirement. In playing with it myself, I could (kinda) reproduce it if I didn't save the buffer. Just to make sure, you're writing the buffer with `:w` before closing?

**Comment by wrightjjw** [comment]
Yup. I usually use `ZZ`, but just to be sure I've tried `:wq` too.

**Comment by wrightjjw** [comment]
I don't think `vim.treesitter.language.get_lang("gitcommit")` does what you expect it to. For me it returns `gitcommit` no matter if the parser is installed or not.

Experimenting a bit, I tried giving it `ada` (which I don't have installed) and it returned `ada`. Putting in `cs` returns `c_sharp`.

I think this function just takes a filetype and spits out the name of the parser, no matter if it's installed.

**Comment by CKolkey** [comment]
Could you try this out and let me know if it works? https://github.com/NeogitOrg/neogit/pull/568

**Comment by cuducos** [comment]
> Could you try this out and let me know if it works?

Worked for me 🎉 Tks

**Comment by wrightjjw** [comment]
Works on Windows too!
